### PR TITLE
acts: add v19.6.0

### PR DESCRIPTION
**Pull request by vvolkl**
and update dd4hep dependency following https://github.com/acts-project/acts/pull/1257

### CODE REVIEW

File: var/spack/repos/builtin/packages/acts/package.py [hunk 1]
```diff
@@ -224,6 +225,7 @@ class Acts(CMakePackage, CudaPackage):
     depends_on("boost @1.71: +filesystem +program_options +test", when="@0.10.4:")
     depends_on("cmake @3.14:", type="build")
     depends_on("dd4hep @1.11: +dddetectors +ddrec", when="+dd4hep")
+    depends_on("dd4hep @1.21: +dddetectors +ddrec", when="@20: +dd4hep")
```
Comment (paulgessinger) [inline var/spack/repos/builtin/packages/acts/package.py:236]: Does this also apply to `main`?

Comment (vvolkl) [reply var/spack/repos/builtin/packages/acts/package.py:236]: Yes, spack has a few special "infinity" versions like `main`, `master`, `develop` that are treated as newer than any numeric versions in comparisons

File: var/spack/repos/builtin/packages/acts/package.py [hunk 2]
```diff
@@ -39,6 +39,7 @@ class Acts(CMakePackage, CudaPackage):
     # Supported Acts versions
     version("main", branch="main")
     version("master", branch="main", deprecated=True)  # For compatibility
+    version("19.6.0", tag="v19.6.0", submodules=True)
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts/package.py]: Is there a reason to change from commits to tags?

Comment (wdconinc) [reply var/spack/repos/builtin/packages/acts/package.py]:
```suggestion
    version("19.6.0", commit="333082914e6a51b381abc1cf52856829e3eb7890", submodules=True)
```

**Review by wdconinc — DISMISSED** [review]

**Comment by alalazo** [comment]
Can you solve the conflict?

**Review by wdconinc — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**